Kconfig.zephyr

```diff
@@ -50,6 +54,18 @@ source "subsys/Kconfig"
 
 osource "$(TOOLCHAIN_KCONFIG_DIR)/Kconfig"
 
+config SYSBUILD_CONTROLLED
+	bool
+	help
+	  This setting is set by sysbuild whenever sysbuild is used.
+	  It must never be selected by anything else than sysbuild but
+	  settings can rely on this to prevent users from configuring settings
+	  which are controlled by sysbuild, for example like:
+	  `bool "Foo" if !SYSBUILD_CONTROLLED`
+	  this line ensures that Foo can only be controlled through sysbuild
+	  when sysbuild is in use. When sysbuild is not in use, then users can
+	  control the setting.
```

Comment:
What would be the point in not using sysbuild but having an application or module set this?

Reply:
when you build a sample without sysbuild.
That could be if you want to have your own multi-image build system, or simply if you prefer to be in full control like today.

Sysbuild is an `opt-in`, just like `west build`.
User should still be able to just run CMake directly for, e.g. `blinky` as:
```
$ cmake -DBOARD=foo -Bbuild samples/basic/blinky
```
in which case they must have all Kconfig settings available.

Reply:
Understood about that but I don't see why they should be encouraged to enable sysbuild Kconfigs if they're not actually using sysbuild. That's setting up for failure if sysbuild behaviour is changed in the future which then impacts their project which is not using system, they can create their own custom Kconfigs for their system.

Reply:
>  That's setting up for failure if sysbuild behaviour is changed in the future which then impacts their project which is not using system,

then the help text might not be clear enough.
It tries to describe what is done in Zephyr like here:
https://github.com/zephyrproject-rtos/zephyr/blob/0da5e417014a86b5cb220e3be2b84c4389ddff1d/Kconfig.zephyr#L869-L870

so that if sysbuild is used, then `MCUBOOT_GENERATE_UNSIGNED_IMAGE` is not available for user configuration directly in the project, as example `blinky`. When sysbuild is used, then this setting must be configured through `sysbuild` configuration.
When sysbuild is not used, then this can be configured directly by the user when building the blinky project.

Feel free to propose changes to the help text, so that this is clearer to any one reading this.

Reply:
Right - I see what you mean, the description makes it sound like users can enable the sysbuild Kconfig themselves not that it is Kconfigs that depend on sysbuild being disabled, will add a suggestion